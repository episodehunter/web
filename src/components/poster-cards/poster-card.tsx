import * as React from 'react'
import styled from 'styled-components'
import { alabaster, melrose } from '../../utils/colors'
import { UnstyledLink } from '../unstyled-link'
import { Navigate } from '../../router/router.types'
import { withNavigation } from '../../router/withNavigation'

type Props = {
  linkUrl: string
  poster: JSX.Element
  topRight?: JSX.Element | string
  bottomRight?: JSX.Element | string
  navigate: Navigate
}

const PosterCardComponent = ({
  navigate,
  linkUrl,
  poster,
  bottomRight,
  topRight
}: Props) => (
  <Wrapper onClick={() => navigate(linkUrl)}>
    {poster}
    <InfoWrapper>
      <TopRight>{topRight}</TopRight>
      <BottomRight>{bottomRight}</BottomRight>
    </InfoWrapper>
  </Wrapper>
)

export const PosterCard = withNavigation(PosterCardComponent)

const TopRight = styled.div``
const BottomRight = styled.div`
  font-size: 14px;
`

const Wrapper = styled(UnstyledLink)`
  justify-self: center;
`

const InfoWrapper = styled.div`
  color: ${alabaster};
  text-align: center;
  font-family: 'Lato', sans-serif;
  font-size: 12px;
  margin: 5px 0 10px 0;
  padding-right: 10px;
  text-align: right;
  border-right: 2px solid ${melrose};
`
