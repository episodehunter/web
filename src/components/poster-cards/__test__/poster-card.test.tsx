import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { PosterCard } from '../poster-card'

it('PosterCard renders correctly', () => {
  const tree = renderer
    .create(
      <PosterCard
        bottomRight="BottomRightText"
        topRight={'TopRight'}
        poster={<img src="some-image.jpg" />}
        linkUrl="some-url"
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
