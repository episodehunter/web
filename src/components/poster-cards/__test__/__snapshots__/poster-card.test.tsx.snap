// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PosterCard renders correctly 1`] = `
<a
  className="sc-bdVaJa sc-bxivhb dTcBvs"
  onClick={[Function]}
>
  <img
    src="some-image.jpg"
  />
  <div
    className="sc-ifAKCX FOUTS"
  >
    <div
      className="sc-bwzfXH uTxCW"
    >
      TopRight
    </div>
    <div
      className="sc-htpNat buBeEl"
    >
      BottomRightText
    </div>
  </div>
</a>
`;
